File: Python/io-example.py
import sys

def process(fin, fout):
    lines = fin.readlines()
    n = 0
    for line in lines:
        print("{:03}: {}".format(n, line), end="", file=fout)
        n += 1

for f in sys.argv[1:]:
    with open(f, "r") as fin:
        process(fin, sys.stdout)

File: C/io-example.c
#include <stdio.h>
#define BUFF_SIZE 1024

void process(FILE *fin, FILE *fout) {
  if (fin) {
    char buff[BUFF_SIZE];
    int n = 0;
    while (!feof(fin)) {
      fgets(buff, BUFF_SIZE, fin);
      fprintf(fout, "%03d: %s", n, buff);
      ++n;
    }
  }
}

int main(int argc, const char *const *argv) {
  while (--argc > 0) {
    FILE *fin;
    ++argv;
    fin = fopen(*argv, "r");
    process(fin, stdout);
    fclose(fin);
  }
  return 0;
}
